### PR TITLE
Flyer Notifications + Org/Flyer Model Changes

### PR DESCRIPTION
**Pull request by vinnie4k**
## Overview
This PR implements flyers notification functionality as well as new changes to the Flyer and Organization models.

### Flyer Model

- Added `categorySlug` field
- Made changes to how optional fields are defined to follow the same format as other models such as `Publication`
- Replaced `organizations` and `organizationSlugs` field with `organization` and `organizationSlug`
- Removed `mostRecentFlyer` and `numFlyers` FieldResolvers as requested by Jennifer
- Removed `nsfw` field in the Flyer model

### Organization Model

- Made changes to how optional fields are defined to follow the same format as other models such as `Publication`
- Made `websiteURL` required as requested by Jennifer

### Flyer Queries

- Rewrote `getFlyersByOrganizationSlug` and `getFlyersByOrganizationSlugs` queries in response to changing the Flyer model to only store a single organization

### Flyer Notifications

- Created a POST endpoint for sending a push notification for flyers
- Created `notifyNewFlyers` and `sendNewFlyerNotification` that sends a push notification to followers via FCM when the organization uploads a new flyer

### Other Changes

- Removed `pull_request` from the prod YAML file causing a new Docker image to be built upon creation of a pull request to `release`
- Rewrote test cases in response to new changes in the schema
- Removed `sendNewFlyerNotification`, `sendNewArticleNotification`, `sendNewMagazineNotification`, and `sendWeeklyDebriefNotification`. Moved these calls to be inline with their corresponding notification function.

## Test Coverage
First, I tested to see if the `followOrganization` mutation worked properly. I have a local MongoDB that contains only one user which is my personal iOS device. I went to the GraphQL playground on my local server and tested the mutation with that single user ID. The database seems to properly update so I tested the POST request and received the push notification. See screenshot below. Note that because there is a new schema, I had to rewrite the iOS code to accommodate these changes.

I then had to rewrite a couple of queries and testing factory in response to the new schema. After failing a couple of times, all test cases finally pass .See screenshot below.

One thing to note that it’s very convenient for me to test the new backend changes since I can directly test it with the frontend.

Update: Making the notification calls inline has been tested and everything works properly.

## Next Steps
- Work on implementing the new mutations for creating, editing, and deleting flyers.
- I am also wrapping up on rewriting the environment variables and creating a Volume backend onboarding document for future Volume backend newbies to reference.

## Screenshots
Screenshot 2023-08-30 at 11 24 46 PM
Screenshot 2023-08-30 at 11 01 35 PM

### CODE REVIEW

File: src/entities/Flyer.ts
Comment (Archit404Error) [inline src/entities/Flyer.ts:32]: Seems like you removed nsfw field but didn't document this in the PR description, in the future remember to document all changes

File: src/repos/NotificationRepo.ts
Comment (Archit404Error) [inline src/repos/NotificationRepo.ts:164]: I don't see a good reason to factor out this logic into a function unless you plan to expose `sendNewFlyerNotification` like the other functions `sendNewMagazineNotification`, etc. are. I know that we also don't use those anywhere else in the codebase, but it seems like we might have been planning to at some point or something.

For now I'd say you can just put the logic of `sendNewFlyerNotification` inside the ForEach and delete the additional function, since it makes the logic harder to follow

**Review by Archit404Error — COMMENTED** [review]
Essentially good to go, just take a look at comments below. Also, as advice for the future I'd suggest making the PR as modular as possible -- I know this is a relatively small PR but it would've been even easier to review if you had split it into 2 PRs, one for adding notifs for flyers and the other for revamping the models you edited.

Once you address my comment about the function that can be inlined I'll approve.

Nice work on the PR description though, I like the detail! You should also rename the PR title to something more descriptive though.

File: src/entities/Flyer.ts
Comment (vinnie4k) [reply src/entities/Flyer.ts:32]: Ah thank you. I knew I was forgetting something.

File: src/repos/NotificationRepo.ts
Comment (vinnie4k) [reply src/repos/NotificationRepo.ts:164]: Looking at the code, it seems like `sendNewMagazineNotification` and `sendNewArticleNotification` are exported but not actually used anywhere. Should we put this logic in their respective `forEach` or do we just keep it as is? Whatever approach we use is fine as long as we keep it consistent.

Comment (Archit404Error) [reply src/repos/NotificationRepo.ts:164]: Yeah let's just inline all of those functions then

**Review by Archit404Error — APPROVED** [review]
Good to go once Joyce approves

**Review by joycrwu — APPROVED** [review]
lgtm! thanks archit for the pr specific comments

**Pull request merged**
